topic-operator/src/main/java/io/strimzi/operator/topic/OperatorAssignedKafkaImpl.java

```diff
@@ -262,11 +262,7 @@ private void executeReassignment(File reassignmentJsonFile, String zookeeper, Lo
                     || line.contains("There is an existing assignment running")
                     || line.contains("Failed to reassign partitions")) {
                 throw new TransientOperatorException("Reassigment failed: " + line);
-            } else if (line.contains("Successfully started reassignment of partitions.")) {
-                return true;
-            } else {
-                return null;
-            }
+            } else return line.contains("Successfully started reassignment of partitions.");
```

Comment:
@tombentley Can you please doublecheck this?

Reply:
I think this is correct to return null. `forEachLineStdout()` will keep reading lines until the given `fn` returns a non-null value. By returning `false` you will end up only reading the first line of the subprocess output. The correct fix is to replace

```
!forEachLineStdout(ProcessHelper.executeSubprocess(executeArgs)
```
with
```
!Boolean.TRUE.equals(forEachLineStdout(ProcessHelper.executeSubprocess(executeArgs))
```